pubsub/producer.go

```diff
@@ -140,29 +178,81 @@ func (p *Producer[Request, Response]) checkAndReproduce(ctx context.Context) tim
 			log.Error("Re-inserting pending messages with inactive consumers", "error", err)
 		}
 	}
-	return p.cfg.CheckPendingInterval
+	return nil
+}
+
+func setMinIdInt(min *[2]uint64, id string) error {
+	idParts := strings.Split(id, "-")
+	if len(idParts) != 2 {
+		return errors.New("invalid i.d")
+	}
+	idTimeStamp, err := strconv.Atoi(idParts[0])
```

Comment:
nit: in future, strconv.ParseUint(idParts[1], 10, 64) can do this directly

Reply:
nice!